internal/model/board.go

```diff
@@ -259,38 +293,143 @@ func (b *Board) MoveCurrentPlayerTo(line, row int) error {
-func (b Board) Size() int {
-	return len(b.Tiles)
+// getAccessibleNeighbors returns the tile neighbors that can be accessed.
+func (b Board) getAccessibleNeighbors(line, row int) Coordinates {
+	var (
+		coordinates = make(Coordinates, 0, 4)
+		lastIndex   = b.GetSize() - 1
+		exits       = b.Tiles[line][row].GetExits()
+	)
+
+	for _, exit := range exits {
+		exitTarget := exit.ExitCoordinate(line, row)
+
+		if exitTarget.Line < 0 || exitTarget.Line > lastIndex || exitTarget.Row < 0 || exitTarget.Row > lastIndex {
+			continue
+		}
+
+		targetTile := b.Tiles[exitTarget.Line][exitTarget.Row]
+		if !targetTile.GetExits().Contains(exit.Opposite()) {
+			continue
+		}
+
+		coordinates = append(coordinates, exitTarget)
+	}
+
+	return coordinates
 }
 
-// BoardTile represents a tile that is placed on a board with a given rotation.
-type BoardTile struct {
+// getAccessibleTilesForCoordinate returns the available tiles from the given coordinates.
+func (b Board) getAccessibleTilesForCoordinate(coordinate *Coordinate) Coordinates {
+	var (
+		results = make(Coordinates, 0)
+		queue   = append(make(Coordinates, 0), coordinate)
+	)
+
+	for len(queue) > 0 {
+		currentTile := queue[0]
+		queue = queue[1:]
+		if results.Contains(currentTile) {
+			continue
+		}
 
-	// Tile is the underlying tile.
-	Tile *Tile `json:"tile"`
+		results = append(results, currentTile)
+		queue = append(queue, b.getAccessibleNeighbors(currentTile.Line, currentTile.Row)...)
+	}
 
-	// Rotation is the tile rotation.
-	Rotation Rotation `json:"rotation"`
+	return results
 }
 
-// Rotation represents a tile rotation on a board.
-type Rotation int
+// GetAccessibleTiles returns the tiles that are accessible by the current
+// player.
+func (b Board) GetAccessibleTiles() Coordinates {
+	accessibleTiles := b.getAccessibleTilesForCoordinate(b.GetCurrentPlayer().Position)
 
-const (
-	Rotation0 Rotation = 0
+	currentPlayer := b.GetCurrentPlayer()
+	if len(currentPlayer.Targets) == 0 {
+		return accessibleTiles
+	}
+
+	for _, coordinate := range accessibleTiles {
+		currentTile := b.Tiles[coordinate.Line][coordinate.Row]
+		if currentTile.Tile.Treasure == currentPlayer.Targets[0] {
+			return b.getShortestPath()
+		}
+	}
 
-	Rotation90 Rotation = 90
+	return accessibleTiles
+}
 
-	Rotation180 Rotation = 180
+// GetSize returns the board size in tiles.
+func (b Board) GetSize() int {
+	return len(b.Tiles)
+}
 
-	Rotation270 Rotation = 270
-)
+// tileId returns  a unique identifier for the tile based on its line and row.
+func (b Board) tileId(line, row int) int {
+	return line*b.GetSize() + row
+}
+
+func (b Board) getShortestPath() Coordinates {
+	var (
+		currentPlayer = b.GetCurrentPlayer()
+
+		size  = b.GetSize()
+		graph = dijkstra.NewGraph()
+
+		// targetNode is the target node, defaults to remaining tile ID.
+		targetNode = size*size + 1
+
+		// reverseCoordinates holds the coordinates for a tile ID.
+		reverseCoordinates = make(map[int]*Coordinate)
```

Comment:
I don't understand the meaning of this naming

Reply:
Done, I added a function to build the graph and renamed this to `getCoordinatesByVertex`